tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc

```diff
@@ -465,6 +465,60 @@ TEST(LinalgBackendEigen, SGMatrix_cross_entropy)
 	EXPECT_NEAR(ref, result, 1e-15);
 }
 
+TEST(LDA, SGMatrix_pinvh)
+{
+	SGMatrix<float64_t>A(3, 3);
+	A(0, 0) = 2.0;
+	A(0, 1) = -1.0;
+	A(0, 2) = 0.0;
+	A(1, 0) = -1.0;
+	A(1, 1) = 2.0;
+	A(1, 2) = -1.0;
+	A(2, 0) = 0.0;
+	A(2, 1) = -1.0;
+	A(2, 2) = 2.0;
+	SGMatrix<float64_t> I(3, 3);
+	linalg::identity(I);
+	SGMatrix<float64_t>B(3, 3);
+	linalg::pinvh(A, B);
```

Comment:
btw you also need to test the other interface where the result matrix is not passed, but generated (here you also need to check that the result has the correct size)